### PR TITLE
Transmition of cleaving through Reindexing_forward

### PR DESCRIPTION
**Pull request by EwenBC**
IMG_20230623_103250
New section in the file Reindexing_forward.v to prove the following property.
If the displayed category D over C is a cleaving, C' is univalent, and the functor F : C -> C' is a street fibration then reindexinig_forward D F is a cleaving over C'.

### CODE REVIEW

**Review by nmvdw — COMMENTED** [review]
I am wondering: is this result original or has this construction been considered elsewhere as well?

**Comment by EwenBC** [comment]
I do hope that it is original but I don't have an extensive knowledge of the Unimath library so maybe I just couldn't find the existing construction.

**Comment by nmvdw** [comment]
It is definitely original within UniMath. Is it also an original mathematical result?

**Comment by EwenBC** [comment]
I don't know I didn't use anything else than the "Displayed Categories" paper to do it but I didn't try to find similar results either... Maybe @benediktahrens knows more about it. (I'm in internship under his supervision.)

**Review by nmvdw — APPROVED** [review]

File: UniMath/CategoryTheory/DisplayedCats/ReindexingForward.v [hunk 1]
```diff
@@ -57,238 +62,238 @@ Notation "¤ x" := (pr1(pr2 x)) (at level 0):reindexing_forward_scope.
...
+  (* Some lemmas to create the theorem to caracterize the equality of reindexing_forward morphisms*)
+  Lemma pr1_transportf_reindexing_forward {a' b':C'} {f' g': C' ⟦ a', b' ⟧} 
+  {x' :ob_disp reindexing_forward_ob_mor a'} {y' :ob_disp reindexing_forward_ob_mor b'} 
+  {Df' : x' -->[ f'] y'} {Dg' : x' -->[ g'] y'}
+  : ∏(p: g'=f'), ↙ Df'= ↙ Dg' → 
+  ↙ Df' = ↙ (transportf _ p  Dg').
+  Proof.
+    intros p e.
+    unfold mor_disp. cbn.
+    rewrite PartA.pr1_transportf.
```
[the remark this answers is not in the document]
Comment (EwenBC) [reply UniMath/CategoryTheory/DisplayedCats/ReindexingForward.v:95]: Yes, I've changed the files imported in the last commit so it didn't before but now it does.

File: UniMath/CategoryTheory/DisplayedCats/ReindexingForward.v [hunk 2]
```diff
@@ -364,164 +369,353 @@ End functor_universal_property_reindexing_forward.
...
-  cbn.
-  etrans. apply (pr2 β a b f x y Df).
-  apply two_arg_paths.
-  - apply uip.
-    apply homset_property.
-  - reflexivity.
-Qed.
+  Proof.
+    intros a' b' f' x' y' Df'.
+    induction x' as (a,(px,x)).
```
[the remark this answers is not in the document]
Comment (EwenBC) [reply UniMath/CategoryTheory/DisplayedCats/ReindexingForward.v:464]: No, I've tried it and coq don't like it.

File: UniMath/CategoryTheory/DisplayedCats/ReindexingForward.v [hunk 3]
```diff
@@ -57,238 +62,238 @@ Notation "¤ x" := (pr1(pr2 x)) (at level 0):reindexing_forward_scope.
...
+  {Df' : x' -->[ f'] y'} {Dg' : x' -->[ g'] y'} (e:↙ Df' =↙ Dg') 
+  : ∏ p: f' = g',
+  ← Dg' = transportf _ e (← Df') ->
+  ← (transportf _ p Df') = 
+  transportf _ (pr1_transportf_reindexing_forward p (!e)) (← Dg'). 
+  Proof.
+    intros p H.
+    rewrite H.
+    destruct p. cbn.
+    apply PartA.transportf_transpose_right.
```
[the remark this answers is not in the document]
Comment (EwenBC) [reply UniMath/CategoryTheory/DisplayedCats/ReindexingForward.v:130]: I should have removed it when I changed the dependencies but it slipped out of my mind.

**Comment by nmvdw** [comment]
Thinking about it more carefully, the construction seems quite similar to https://github.com/UniMath/UniMath/blob/master/UniMath/CategoryTheory/DisplayedCats/Examples/Sigma.v#L75. However, instead of composing two displayed categories, you compose a displayed category and a functor.

File: UniMath/CategoryTheory/DisplayedCats/ReindexingForward.v (hunk 3 above)
[the remark this answers is not in the document]
Comment (EwenBC) [reply UniMath/CategoryTheory/DisplayedCats/ReindexingForward.v:130]:
```suggestion
    apply ransportf_transpose_right.
```

**Pull request merged**